version: '3.8'

services:

  identity:
    image:  casbin/casdoor
#    build:
#      context: ./casdoor
#      dockerfile: Dockerfile
    restart: always
    ports:
      - "8000:8000"
    networks:
      - backend
    depends_on:
      redis:
        condition: service_started
      db:
        condition: service_healthy
    environment:
      RUNNING_IN_DOCKER: "true"
    volumes:
      - ./config.local/casdoor:/conf/

  db:
    #db with dev data
    #image: app-postgres12.2:0.0.1
    # for pure postgres
    image: postgres:13-alpine
    volumes:
      - postgresql-data:/var/lib/postgresql/data
      - ./config/db:/docker-entrypoint-initdb.d
    restart: always
    environment:
      #PGDATA: /var/lib/postgresql/data
      POSTGRES_DB:        data
      POSTGRES_USER:      postgres
      POSTGRES_PASSWORD:  1gjcnuhtc1!
      POSTGRES_MULTIPLE_DATABASES: data_rotated, reference, identity
    ports:
      - 5400:5432
    networks:
      - backend
    tty: true
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 10s
      timeout: 5s
      retries: 5

  redis:
    image: redis:6.0
    restart: always
    volumes:
      - ./data/redis:/data
    ports:
      - 6379:6379
    networks:
      - backend
    tty: true

volumes:
  postgresql-data:
    driver: local
networks:
  backend:
